### PR TITLE
Fix UI

### PR DESCRIPTION
**Pull request by zelfroster**
Since, There should be different options for the Appearance tab in webui. I needed to fix the typography first and make the UI a bit consistent.

I will do this bit by bit, but first I am going to fix the things which are most important and make sure that they look good enough and are usable.

Likewise, I have defined the typography and took inspiration from [tailwindcss-font-sizes](https://tailwindcss.com/docs/font-size)

And this is the homepage of webui which I have redefined.
image

Please give some feedback @csoler @defnax

### CODE REVIEW

**Comment by zelfroster** [comment]
This is the Network tab where the content is scrollable but the heading remains fixed so that the user can search bar remains visible. And I will try to do this with every section with a consistent UI.
image
image

**Comment by zelfroster** [comment]
This is how People tabs look right now.

image
image
image

**Comment by defnax** [comment]
im not fan of big buttons i like more how on professional web mails with small buttons :)
image

the subject field is on wrong place :)
image

**Comment by zelfroster** [comment]
> something in mails is broken? it does not show more the top field and columns
> 
> image

For now, everything else must look broken, which I have not fixed yet. So, just some more time needed.

**Comment by zelfroster** [comment]
Made Download UI a bit better. But `getChunkStrategy()` is not yet implemented, so I was trying to implement it myself today, I think I would be able to do it.
image
image

**Comment by zelfroster** [comment]
Fixed chunk strategy in downloads by implementing [`getChunkStrategy()`](https://github.com/RetroShare/libretroshare/pull/103) and adding new method to load chunk Strategy separately.

image

**Comment by zelfroster** [comment]
I have also fixed the Chat UI overflow for now.
image
image

**Comment by zelfroster** [comment]
Fixed Channel Layout, Post Layout, and UI changes.
image
image
image

**Comment by zelfroster** [comment]
I fixed the layouts for boards, but while fixing layout for forums, there was a bug which was causing webui to crash (a huge number of api calls). So, I will fix the layout after fixing the bug in Forums in the 2nd phase of GSoC (Improving Forums is already in my GSoC timeline).

Now, only mail layout is remaining to fix now in this PR.

**Comment by defnax** [comment]
i has tested it not had time test on longer time
looks more clean then before, i cant read my mails often it takes long to load.

i cant view attachments its hidden or my notebook screen it to small? i can click on up down on scrollbar then i see the attachments
image

**Comment by zelfroster** [comment]
> i has tested it not had time test on longer time
> looks more clean then before, i cant read my mails often it takes long to load.
> 
> i cant view attachments its hidden or my notebook screen it to small? i can click on up down on scrollbar then i see the attachments
> image
> 

Ahh, I tried to make attachment section just like in the qt ui, fixed at bottom.
But since I primarily use Firefox, it has thin scrollbar so its visible on my screen, I think I need to optimise it for smaller screens.

Thanks for testing it out. I will adjust it so it will be better to see attachments.

**Comment by zelfroster** [comment]
It should now show it correctly, @defnax 
image

**Comment by defnax** [comment]
thx, i will test it tonight

**Comment by defnax** [comment]
on myscreen chrome with 100% zoom (15,5 inch screen)
image

75% zoom
image

**Comment by zelfroster** [comment]
I work on a desktop with a 24-inch screen, so that's why I am able to see clearly 😅.
I am thinking of an appearance setting where we can set the font-size accordingly, and It will make it easy to view on smaller screens comfortably.

**Comment by zelfroster** [comment]
> i think i know why it takes so long with mails it loads 500 mails to get view my mails takes up to 30 seconds
> it needs this feature
> View only 50 mails per page no endless loading /  not load all mails (low)
> image
> https://github.com/RetroShare/RSNewWebUI/issues/69

Got it 👍🏻, so now what we need is pagination.
I will try to implement it in another PR.

**Comment by csoler** [comment]
probably the transfer of mail data between libretroshare and webui is the bottleneck here.

**Comment by csoler** [comment]
it's a bit weird on the user's point of view that you can change the chunk strategy but you cannot really see which chunks are available or not. Would it be complicated to draw the progress bar using colored chunk (blue/red/yellow) as we do in Qt? I suppose one major challenge is to make it appear clean whatever the scale. Qt does the filtering for us, but maybe JS will not.

**Comment by zelfroster** [comment]
> Would it be complicated to draw the progress bar using colored chunk (blue/red/yellow) as we do in Qt?

I think yes it's going to be complicated, but I will have to see what data are we getting from the `FileDownloadChunkDetails()` endpoint, which I suppose returns the individual chunks for a file being downloaded.

**Comment by csoler** [comment]
you'd need to use the list of chunks (a table) in FileChinkInfo structure, especially the member "chunks".

**Comment by zelfroster** [comment]
> you'd need to use the list of chunks (a table) in FileChinkInfo structure, especially the member "chunks".

Yup, I tried that endpoint, This was the response for a file.
1687959718_Jun28_19:11:58
1687959736_Jun28_19:12:16

What does 0, 2 etc. stands for? I will look it up in the libretroshare.

**Review by rottencandy — COMMENTED** [review]
For mails I think we could still attempt clientside pagination, since the main bottleneck is likely because we are fetching mail details for every single mail  in the entire list up front when we render it.

File: webui-src/app/scss/components/_posts.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+.posts {
+  height: 100%;
+  margin-top: 1rem;
+  flex-direction: column;
+  overflow: auto;
+  &__heading {
+    display: flex;
+    justify-content: space-between;
+  }
+  &-container {
```
[the remark this answers is not in the document]
Comment (zelfroster) [reply webui-src/app/scss/components/_posts.scss:12]: Yup, I also tried to follow BEM convention, but I never really deep dived into it 😅.
So, It looks like a mix-up of names. What do you suggest doing here 😅?

File: webui-src/app/mail/mail_resolver.js [hunk 2]
```diff
@@ -152,8 +138,28 @@ const Layout = {
...
+          m('.top-heading', [
+            m(
+              'select.mail-tag',
+              {
+                value: tagselect.showval,
+                onchange: (e) => {
+                  tagselect.showval = tagselect.opts[e.target.selectedIndex];
+                },
+              },
+              [tagselect.opts.map((o) => m('option', { value: o }, o.toLocaleString()))]
```
[the remark this answers is not in the document]
Comment (zelfroster) [reply webui-src/app/mail/mail_resolver.js:154]: Ohh, I thought it is used like a fragment, and then forgot the usecase of fragment.
We only need it when there is no parent, got it 👍🏼.

**Comment by zelfroster** [comment]
Also, Eid Mubarak @rottencandy 🫂☪️

**Comment by zelfroster** [comment]
@rottencandy If you have time then please look into this code, It's not working correctly.
Even though I am trying to trigger a re-render manually, It is not working.
I am not able to figure out what's the issue as it should work. Please help.
re-render-issue

**Comment by rottencandy** [comment]
The reason it's not working is because we're using `popupMessageCompose` which uses m.render() to show the modal. It draws outside of the main mounted tree and will not respond to redraw calls or state changes: https://mithril.js.org/render.html#differences-from-other-api-methods

We have this same issue with other modals as well, and we should really stop using m.render and switch to creating modals inside the main tree soon.

For now I think the only way might be to create a new route & page for composing : (

File: webui-src/app/scss/components/_posts.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (rottencandy) [reply webui-src/app/scss/components/_posts.scss:12]: Yes normally - is used just for separating words in the same block name, and __ is used to separeate blocks.
So whenever you create a new sub block section prefix it with __

**Comment by defnax** [comment]
oh as page the mail composer will look unprofessional

**Comment by rottencandy** [comment]
Alternatively we could use datalist as a temporary workaround

https://stackoverflow.com/a/19779010/7683374

**Comment by zelfroster** [comment]
> Alternatively we could use datalist as a temporary workaround
> 
> https://stackoverflow.com/a/19779010/7683374

At first, I had tried datalist only, but it's the same issue.

**Comment by zelfroster** [comment]
> We have this same issue with other modals as well, and we should really stop using m.render and switch to creating modals inside the main tree soon.

I think, this is something I can do, I will give it a try. Thanks for helping 😃

File: webui-src/app/scss/components/_posts.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (zelfroster) [reply webui-src/app/scss/components/_posts.scss:12]: Got it 👍🏼

**Comment by zelfroster** [comment]
I think everything is fine in this PR now. @csoler

**Comment by zelfroster** [comment]
I tried to move the `modal-container` inside the main div and changed m.render to m.mount. But it still doesn't work.

I had also tried this guide https://mithril-by-examples.js.org/examples/modal-2/ , but no success.
the state doesn't update [tried on flems.io](https://flems.io/#0=N4IgtglgJlA2CmIBcBWADAOgBwEYA0IAzvAgMYAu8UyIA8gG7wBOsAhgJ4YBWhIBpAewB2hAQhqDYbAA7FqBQqSZjYyANpo8aALoEAZhAS8ka0ENZhESEBgAW5MKv7DKQ8jRABfPGYtW6jCwc3LzObvBuNAD0AFQxADpCACq2EIQABAzMbOzpgmDSwhHk6dLK9NDwGazpegCuUgC0ikzwEXkCgekk8JZuGIkAkiVp6WACdeFQ6eQC6TXE0qxMrJTpAGpRACK0ALIzrfDz0tIRUFQzc+S2RwBGAlCciQDCqbBQrULphHUnsBAXWaZQI5eatdKfc6taYQNxXVIZciHAbJG7A7IcDoFIpuPKsL63I5CKqUab49gAdxu4IpEGusPS7AmTGO0gA5IREhV4BT0rc6iVaVI+UdIcxAXMoGlpKDOsxZtJuoF2PShABzbqwYgomJRRKJQQiEpZIK5AC8tUmFAgwgAFABKdLAfXkBAlKACMAut15N4fCL6oTxciGwjGkEcZ4uViw5jpC3OoPkYPcilIdIO+MAPl9hn9QaTnkDwda5DqTC+ieDweESngq3gtvojqrybbpD9n3j6XoGA7ec+Lur5CiUXSAEETmd5h03DHiSygdcjtOBHo+Q92EO2x6wN2PaQ6n1yH3Wg2AKIIY+2tlS+hs+3b4O7vtsQiEAByfm7bLlprZT7kAeR7FBg9yPBgrBTkIUCvHmtq7o+XzpChKGjukuwTLiCzwEsKxrOsOz7EibTpNS8CAWAGDjJM5AIZ6eDoqaUZzrGTBIcO3jbsIhJ6AIrS-FADZNi2gH9u8XYWr24n5tuXFJjWQi8fx4rjIwmatsO6HjjAM6kG+GS0tctSsOcjQTCU8AAB50vMQgQGAqw2gWw7Pp6r6sO+AAyaQnqZUA3qk5wPoBpbll8xK8gACsokDELaLJmjmmltm5VH+eejBuD5YYRMwN74g5Tk4lAbKMexgGeBxbbycOwitGpjaiQpbboVsJLKLky7pH+oK7gcbQYOkACqhCwhqVE0biRm2IB6FCA0sAHBAapqsw1RCLk4xCUtMldg1nSsEttgCAIADWhAoq5DjUVhdG7oxC1SNVw7AceYGbhgB2MHB7z0WAL3kLVbaphpRaFoknh8EQPTWsIxggAATAALEgADMaNeD4IDmJYNCYTtITQ4arjuNYsQJKiowE0dWKFMSuJ1MQiJoiaoL5PTxSXKU5SVPMiT1MKLRtF8ghdPi0xikw1TpFKR0CGqjT-GdRy0lAa2CnSqRfNcqyJBSMbkBkfGLmiTNxkC+kQKQZ0zvy5CzEIQ3jhhDy07CYb4qQRykPiiSEukxK5WSm1UuK6QzQyTLlqyHI9gCvISxHhhLQHUtUIkwiXIqa5KswKraxqJDaoG55WacFAXKsSLVOCnm1CoAgUoQSCBrClBMHorDexODvS06iQoeQdIIOmVG-bJyEk8UY8YBPg7Ifbjst060DpmGTDjYxlBWeQ69IuNnhqNog89UIzywAI8Vr98B-qva6b0AI0CQ4G44lAgnklEUfICo73wnQaNMAOZROiVGmAIZcTBaTEBmORfWKcRSBzmAbLqcx9JXyON1baR0dR6gLKGEoNMloWnqEIOGQgRID2Qj6K2NsqCDCgKfU+YUKzUNQvHHktpgDV2lumYAw9XTwEYtPNwjEl7CBEXdRiwgL6YM8FVdhHD0gQHXLaOhKsoCMPSAAQgtE9WAzVlGoXQgAdSJG0aYQIjqX15N1NmmIOY4kslZeAh41h0k5MhYx6FrK2UKo5YewghoACE3GsHNiokYMsZI9XXPSDITiGbkDwKfHxY5bJCiWkISBtQIA0huDrBEdNnEqIyN9KgKJjFoTHIMI09YoCMQpBYiUEJehyiiXk6WJRlLggSSU5JaTlHoTVAIKo8xUFXVSq1McpjCl2z-lnUYGiqBNKOJ6Wy9iIy5CScUIZHD0J6FinA6mbtdqek5m4OOxJd6cN5FLJpqQOzJykPskxY4D6rTjA49gcceLwBNqpDpJ1zpVOqawiKi03ng2MbQ7sghaIAG43kQrGLaH5qTvHKMTNU1C9V2nqSMbi956RTGGwbqbdZ2y47fVpgE4qOt0EXLdPAMFxL0joRSO0ZpXT1kxzwsUOOvspC3G7mdR51tbAvNgG89JPMqhHlYLcWAuQKknIyMQgZoFZXKJihs7UrRRCwEJRgZclDMxJSUeylCsjL7xRWVoqASFrWoSotCFYFIHQ6o4VVb16QYXVPkrisAN5bo7TKn6kNbJbBozKjMEe8B7SYvZVGsNR1GiiPIHGzNSbI2hpwbARoEiRARqxcGm8xbS0upQji6tQ92CnHTGyStya62yzSEqhAUB0wOu0XowOi1W11trP8G2Gk-W4t7dMC0xaMAvzLdagNLqg3VtnTvZMC7jG5s3coqNLanQjutmddMFrkp5AANTnsRbOI1rLL5qnUfaa9CjGIAAMAAkwBSCeDyKOi6r7nXEu3biwDHDHTKKGQG8GUMFCwyCSIGgaNUY4AAJxY18HjawRMwikwkPDEok5FSkKtPB8dND4AlEICdCkmrSFHWIKfWhd1uxoGRchVFtaUKphPY6S1qaoLsiHa6wKOA41sk1e1cYD4hMoSjYJ9IbIL5HpFLY7mAhThfBqAW6Tby5NVuqXuxZQh9O4s48S8gDb4BNpbROs+GiyN1qo03WjBw6gUR3T6v1K7cVsloBp124adXAeqehTK7RrgTDVFK-pBaymBxJBcSOXxmQThOD2B6Or0KZMQQJ+sLIs4KjiXnJgBdxqamIIxXLMELiZY+fCFcMEitvS5uBJ4O6nM0bOekAAZN1tFxDGJmeMYI0eCmAASJBL4BaOjokz1TM1NqUzbOBdwjMqabtzDB2o5vGOLfwm+bItvwDjeuxbdrjvefm3dHt0i-W2swbaaARL2UdZc13LU7n2VLuUb6hdoH0iAeg4GSad16KHneq1xihGkLQ2IGQeDCMUZIERpjbwGH-B9nfMTFwxQaCtaUY5JgapYTplY6fJYMBxqk7YyhPibhGhd0gCq9e+JCDNGYKomnDd6djQAF5WfSDgAAbNIKybHwaJEgml2tFOpTqnTIjXo4vAwAAEVbsCOX4Y2pl4CNAZLWo5noD1LFIHSdg1OA1AmAOp7uZv0w4ERUDgsav4Aa5WJYbXZkLJKMN3ua3Ju7eC8d6fK3NvTcWYt6-AsGBeqYhl1fOkzl0wGFcVALnCpqeMYBXvdIZPkJq2uPbtAaAACkXObgrXsEX0vXPRU2zVMoSYUAM1iH4umInorbSIxQCgPA3fe-960BgAA7CgJ9p8pSEBlBwZPCAxeny4EzYeeh2At-CG4HtxRmBc6OitIQevKBgBXt7cITAd-2UCUnkyZkGSYBQIQZXTtY+cCCkcWt9L4PJ51+ZAUueMD38f0l1i1rXXUaF3zVCEE31Py51lyp3SEVz3GRl6HSDRmQKQLAFrzFQbywmb0kDb3SAAGIAAxEg2vAQKyZoWwUyJuUnf-ZAu-ZAhAwglDFgsgpgKEdMAg54bgwXUXb4MQaAQAp2AtItIzDIWtQnYnffDPQXJXFCKPNNQtYtDIYtAnZYKQ0nP-FAJXKPWHODZyBGFAHAJANALwXQEAf4IQC6dQdHGgSAa4TeJwEAcsVQawewcgWQJAUcSYaQM6NUPsT0KIew2wRwlXHADAcIoXIIrWRw7DEACzU4GgFoCAaQdwWDeHAwmgHAYfLAEwswzwIAA), the count doesn't update.

**Comment by zelfroster** [comment]
> I had also tried this guide https://mithril-by-examples.js.org/examples/modal-2/ , but no success.

It creates and attaches a vdom node to the dom and creates the overlay container which contains the modal container. But still redraw doesn't seem to work.

**Comment by rottencandy** [comment]
Ah if you're moving the modal div inside the main div you can't use a second m.mount, since it is part of the parent mount it should just be used as a regular component. This would require reimplementing the create modal handler and components in a way that doesn't use another mount or render.

**Comment by rottencandy** [comment]
For the flems example you would need a component variable. [updated code](https://flems.io/#0=N4IgtglgJlA2CmIBcBWADAOgBwEYA0IAzvAgMYAu8UyIAsgPZQCGsGAVoSAafQHaH0ENHrFhMADsWoFCpAE6DYyANpo8aALoEAZhAScky0LyZhESEBgAW5MEu59KvcjRABfPMdPmQAeQBu8HJiAJ7snA7O8M40APQAVPEAOrwAKlYQhAAEAUGhWTxg4nzR5FniCv7Q8NlMWdoArqIAtLJy8NEF9IFyWSTwZs4YKQCSZZlZYPQNUVBZ5PRZdcTiTHJMlFkAarEAIr608+3wS+Li0VBU84vkVicARoxhKQDCGbBQ7bxZhA1nsBArgscj18msTl9Lu05hBnDcMtlyMdhmk7iC8kwQl0iiVnAUmN97ideDVKHMCSEAO53dpZSkQW6wrIhaa9CTiADkhBSVXglKy9waZXpogFEIuQSBiygmXE+W6QQW4j6PRCjN4AHM+rBiCj4rEUikePwyrlgpisgBeeozCgQPgACgAlFlgIbyAgylB6GB3Z6Cu9PtFDbwkuRjYRTaDMS9HExYUEra73WHeZSkFlnVaAHwBvRB0Oh8huENh9rkBpyb5uothvjyeAbeAO-wumthjvhwNfJP+DCkbvB2vkMOxWJZACCZwuSy6znjJN6wNuJxn9G0AqeKZH5G9YCT3tIDUG5H77SbAFEECeHRyZf4OU7t2G9-2xIRCAA5bxJjkK80hByz67vQR4nhgjxQGE7IXG8+YOnuT7fFkKEoWOWQMDMZTLPAqzrJsWz7IcSIdFkNLwMBYAYFMWEIT6eDogBsbzgmchIZ2HjbnwRLaPQ7R-MwlAtm2wEDvmPbWn2YkfF826ccO3HwLx7TtFMgRZu2O6juOE4wLOpDvtk9K3PUTCXM00xlPAAAeDJLLwEBgBs9qFp2L4+m+TAfgAMpkp5mVAt4ZJcj7AeWlbfCS-IAAoKJAxAOr0lq5ppbkgVRAUXoEzi+ZG0RBLeBKOc5uJQByDFscBbjsTu8mdnwqkKhpwHobspIKFiK5ZP++R7kcHQYFkACqhCwlqVE0XixlWC1468E0sBHBAGoakEtS8FiUzMIt0kFlkjX+CwZH0PQADWhAomlE3TM4dFgAx82iDVnaHsepQQU8GAHfAcEfHdz3Fng25phpJZFmDbhcEQ-R2nwBggAATAALEgADMqPuJ4IAmGYNAMNt4RQ8aTguBYCTJKiEz40dhTFCSeINMQiJoma+S07iZTAhU3TVEsKSNKKbQdN8PA9PZcyQmts4yiw9Aas0AKnSc9JQKtwoMhk3y3BsKSUvG5DZMp8xoozibAgZECkKds6CuQCy8INE4YYwR2wpGBKkCcpAEikRJZCSeXkht1KSnSGtMiylanJy2RpuLYein7kvQikfDXMq64qkEaqa1qJC6iGF7WecFBXBsSK1LSXn1Io9CUoQSAhrClByNoTCe5OdtyNkNYoeQDIIBmVG-QWKQocTpRDxgI+ychtv2w3rrQBmkZyGNDGUNZ5Ar0iY1uMoGhj91vAvLA9AJcvPy75qToZv49DQCkYMpBOZQIF5ZQlAKQr2z8VjTB8MU5RKjVDmPQFcch6TEGNpKXWehFp+14IsPWnVFgGXPicLqW0WB6gNIWCMZRqaLWtI0XgsNeDCWTMhf0FsrZUBGFAAA3EfGhvYMDl27v2G65BmHISPuFKsVDUJZBBsADhi9gD9w9PABiE9nAMXnnwBifBT4YLcNVIRwisgQA3A6WhSsoAMKyAAQmtI9WAIlkJaKyOhAA6sSDocxgQsDPvyLqrMLTs3plZay8AjybAZNyKxWj0I2TskVJy-c+CDQAEJ+KYKbbR4xah5kAZnRk2QvGlCBsE4R6E7IikWkgsouhaQhy1gibEdNSjaOyN9KAKJrFoXHCME0jYoAMUpA4qU+0BgKiSfUCA3cyg8T4pgypWTnBH2sehDU9AahLBQZdLSO47F3G+HURR3wJj6KoJ0k4Po7LuOjFiSZI5cmoXQtoOKxsqYux2j6apzguT+xsmUOOydOkZAHAnWA0yQnjl3itRMHjALZEUspSUakTj-zOo0ppAjIoLX+ShMG1jEWTAdKCnJTTNG4oan09SljcV5PHLY-WNclxoh6piF5B0joRJKlrNBjzPTwHhSSmx450idC6YM2krJyjgmeZklx9x26nS+ZbKwvyUVNPQhUGox4mD3FgFib6tzshEKqRzDlJLYqHN1O0AQsAiUYBXBQrMyU8WcuPqohKuzDFQCQra1CVFoTrEpM6OV1jqo+tQmipp8lcVgFvNRe55V-UoVDRyKwqNyrzAHvAJ0OKSUxvDdtZocjyAJuzSmqNmKOQZpYM0LZhBI0XOsTGrZFbXUoV7nWvuIRzgZg5DW1NjaZSEBVQgKAGZHVGNMf7BaHa631gBFbZqlbbUDrmNaLZGBH7TtxYG11wa60Ls3uc11+bl1utvO21047LanQzFalKBQADUl6sjqIYgAAwACTAFIG4AoE7zr3pdSS3duLv3CJdFo-5gaIZQ2IGQKJ-AaAADZUZICRigTGXhcYWEJpEEmwg4ZlCnMqEhtpINTpQv6Qg-9KTapISwYgLD4BlB4FhJMaBeEoQxQ2lCaYz0umtem9ktatExqsDgBNHJtVtSmI+Ud0bbycgYhyU+J6xSuOuN1c4GzJgRt-XxqTvGq0Hp-nwbTTTWMkvIM2+Arb20FpQseydxK60kbruRo4DQKJ7tRQW9duKOS+BU87baQFp0aZmeOLKnRbjTA1DKjJantq1NeYHMO6pupRxwyIvcEnLnjgKfA6OjZehpyVN1Dc8BVTqjzjqGROXeCXDmP69Cy40Rrg3K9E8m4oJ6uEfZsj9ysgADIeuYqIQxIz1ipGDyyByAAEiQM+vmWDGIM9Y7Nra5NWxgd-O2aciSKfNmfXUC2tFbIzMAS+HJ0HEATVu5bu34Acg800ujzh+3cPS8IlR12HTQFs5yzrjm27lfc-6v1Fz-1ZG-RDEM11aLNfepBEIDEcNITAzDSD8NkZIARhjDwyGfD9g-ETRwpQaCw80U5OQGpYQZkY0fVYMAxqU6YzXZwzQ26QDVSvAkhBWhBB0Qz3iTPRoAC8zNZBwNB8Q1leHPwduyTRNOZSagzAjAYkuQwAAElYhGud4Q2Zl4DNCZA265Poj2rFIAyEI9PA3AmAPQU35uMw4EYeDws6v4Ca-WGYHX5lLKaKN-uG3duTMO6d0fa3tv2726yIx53KQMA0qxA24oo1IMZl0L4phofbf0+o9obeUeGcq1uA7tAaAACkDO7jLRsMXsvDPxVWw1AoGYUAs2CD4hmMn4qHQIxQCgPAPe+8D-UBgAA7CgJ0DOu1ykxKnhAEuj5sEZv3bQIRW9REewUUoQQGcsGWrwfXlAwCL09lEOQO+HKRJcqn3X+vviYBQIQFXDt4-WF5g2xlKfTLe6FFHjAD+n+x7YKLQNpbrNC74ai8D9pb5n7U4BR05ZBK77hIwDBZCowoHIFgB14SqN43Qt4iDt5ZAADEAAYqQXXvQNZK0FYGZHXJTn-igffigYgUQQAJxsHkFyBQgZiEEvC8Ei7i4-CCDQAAEOxAGlp6b8Ak5rDk775KgO7K4oRPyAH3LiEbaSFbJSFk4U5R6-4oDK5KG8BI4QYuTwwoA4BIBoDuBaAgAAi8DnQqDY40CQC3Brz2AgCVhKAWA2DkCSBIBjgzDiCnQahcJgCxDOFWCuGq44AYDRHQZhEayuFoYgAmbnA0BtAQDiAuAyDI4mE0A4Aj5YAWFWFuBAA)

**Comment by zelfroster** [comment]
> For the flems example you would need a component variable. [updated code](https://flems.io/#0=N4IgtglgJlA2CmIBcBWADAOgBwEYA0IAzvAgMYAu8UyIAsgPZQCGsGAVoSAafQHaH0ENHrFhMADsWoFCpAE6DYyANpo8aALoEAZhAScky0LyZhESEBgAW5MEu59KvcjRABfPMdPmQAeQBu8HJiAJ7snA7O8M40APQAVPEAOrwAKlYQhAAEAUGhWTxg4nzR5FniCv7Q8NlMWdoArqIAtLJy8NEF9IFyWSTwZs4YKQCSZZlZYPQNUVBZ5PRZdcTiTHJMlFkAarEAIr608+3wS+Li0VBU84vkVicARoxhKQDCGbBQ7bxZhA1nsBArgscj18msTl9Lu05hBnDcMtlyMdhmk7iC8kwQl0iiVnAUmN97ideDVKHMCSEAO53dpZSkQW6wrIhaa9CTiADkhBSVXglKy9waZXpogFEIuQSBiygmXE+W6QQW4j6PRCjN4AHM+rBiCj4rEUikePwyrlgpisgBeeozCgQPgACgAlFlgIbyAgylB6GB3Z6Cu9PtFDbwkuRjYRTaDMS9HExYUEra73WHeZSkFlnVaAHwBvRB0Oh8huENh9rkBpyb5uothvjyeAbeAO-wumthjvhwNfJP+DCkbvB2vkMOxWJZACCZwuSy6znjJN6wNuJxn9G0AqeKZH5G9YCT3tIDUG5H77SbAFEECeHRyZf4OU7t2G9-2xIRCAA5bxJjkK80hByz67vQR4nhgjxQGE7IXG8+YOnuT7fFkKEoWOWQMDMZTLPAqzrJsWz7IcSIdFkNLwMBYAYFMWEIT6eDogBsbzgmchIZ2HjbnwRLaPQ7R-MwlAtm2wEDvmPbWn2YkfF826ccO3HwLx7TtFMgRZu2O6juOE4wLOpDvtk9K3PUTCXM00xlPAAAeDJLLwEBgBs9qFp2L4+m+TAfgAMpkp5mVAt4ZJcj7AeWlbfCS-IAAoKJAxAOr0lq5ppbkgVRAUXoEzi+ZG0RBLeBKOc5uJQByDFscBbjsTu8mdnwqkKhpwHobspIKFiK5ZP++R7kcHQYFkACqhCwlqVE0XixlWC1468E0sBHBAGoakEtS8FiUzMIt0kFlkjX+CwZH0PQADWhAomlE3TM4dFgAx82iDVnaHsepQQU8GAHfAcEfHdz3Fng25phpJZFmDbhcEQ-R2nwBggAATAALEgADMqPuJ4IAmGYNAMNt4RQ8aTguBYCTJKiEz40dhTFCSeINMQiJoma+S07iZTAhU3TVEsKSNKKbQdN8PA9PZcyQmts4yiw9Aas0AKnSc9JQKtwoMhk3y3BsKSUvG5DZMp8xoozibAgZECkKds6CuQCy8INE4YYwR2wpGBKkCcpAEikRJZCSeXkht1KSnSGtMiylanJy2RpuLYein7kvQikfDXMq64qkEaqa1qJC6iGF7WecFBXBsSK1LSXn1Io9CUoQSAhrClByNoTCe5OdtyNkNYoeQDIIBmVG-QWKQocTpRDxgI+ychtv2w3rrQBmkZyGNDGUNZ5Ar0iY1uMoGhj91vAvLA9AJcvPy75qToZv49DQCkYMpBOZQIF5ZQlAKQr2z8VjTB8MU5RKjVDmPQFcch6TEGNpKXWehFp+14IsPWnVFgGXPicLqW0WB6gNIWCMZRqaLWtI0XgsNeDCWTMhf0FsrZUBGFAAA3EfGhvYMDl27v2G65BmHISPuFKsVDUJZBBsADhi9gD9w9PABiE9nAMXnnwBifBT4YLcNVIRwisgQA3A6WhSsoAMKyAAQmtI9WAIlkJaKyOhAA6sSDocxgQsDPvyLqrMLTs3plZay8AjybAZNyKxWj0I2TskVJy-c+CDQAEJ+KYKbbR4xah5kAZnRk2QvGlCBsE4R6E7IikWkgsouhaQhy1gibEdNSjaOyN9KAKJrFoXHCME0jYoAMUpA4qU+0BgKiSfUCA3cyg8T4pgypWTnBH2sehDU9AahLBQZdLSO47F3G+HURR3wJj6KoJ0k4Po7LuOjFiSZI5cmoXQtoOKxsqYux2j6apzguT+xsmUOOydOkZAHAnWA0yQnjl3itRMHjALZEUspSUakTj-zOo0ppAjIoLX+ShMG1jEWTAdKCnJTTNG4oan09SljcV5PHLY-WNclxoh6piF5B0joRJKlrNBjzPTwHhSSmx450idC6YM2krJyjgmeZklx9x26nS+ZbKwvyUVNPQhUGox4mD3FgFib6tzshEKqRzDlJLYqHN1O0AQsAiUYBXBQrMyU8WcuPqohKuzDFQCQra1CVFoTrEpM6OV1jqo+tQmipp8lcVgFvNRe55V-UoVDRyKwqNyrzAHvAJ0OKSUxvDdtZocjyAJuzSmqNmKOQZpYM0LZhBI0XOsTGrZFbXUoV7nWvuIRzgZg5DW1NjaZSEBVQgKAGZHVGNMf7BaHa631gBFbZqlbbUDrmNaLZGBH7TtxYG11wa60Ls3uc11+bl1utvO21047LanQzFalKBQADUl6sjqIYgAAwACTAFIG4AoE7zr3pdSS3duLv3CJdFo-5gaIZQ2IGQKJ-AaAADZUZICRigTGXhcYWEJpEEmwg4ZlCnMqEhtpINTpQv6Qg-9KTapISwYgLD4BlB4FhJMaBeEoQxQ2lCaYz0umtem9ktatExqsDgBNHJtVtSmI+Ud0bbycgYhyU+J6xSuOuN1c4GzJgRt-XxqTvGq0Hp-nwbTTTWMkvIM2+Arb20FpQseydxK60kbruRo4DQKJ7tRQW9duKOS+BU87baQFp0aZmeOLKnRbjTA1DKjJantq1NeYHMO6pupRxwyIvcEnLnjgKfA6OjZehpyVN1Dc8BVTqjzjqGROXeCXDmP69Cy40Rrg3K9E8m4oJ6uEfZsj9ysgADIeuYqIQxIz1ipGDyyByAAEiQM+vmWDGIM9Y7Nra5NWxgd-O2aciSKfNmfXUC2tFbIzMAS+HJ0HEATVu5bu34Acg800ujzh+3cPS8IlR12HTQFs5yzrjm27lfc-6v1Fz-1ZG-RDEM11aLNfepBEIDEcNITAzDSD8NkZIARhjDwyGfD9g-ETRwpQaCw80U5OQGpYQZkY0fVYMAxqU6YzXZwzQ26QDVSvAkhBWhBB0Qz3iTPRoAC8zNZBwNB8Q1leHPwduyTRNOZSagzAjAYkuQwAAElYhGud4Q2Zl4DNCZA265Poj2rFIAyEI9PA3AmAPQU35uMw4EYeDws6v4Ca-WGYHX5lLKaKN-uG3duTMO6d0fa3tv2726yIx53KQMA0qxA24oo1IMZl0L4phofbf0+o9obeUeGcq1uA7tAaAACkDO7jLRsMXsvDPxVWw1AoGYUAs2CD4hmMn4qHQIxQCgPAPe+8D-UBgAA7CgJ0DOu1ykxKnhAEuj5sEZv3bQIRW9REewUUoQQGcsGWrwfXlAwCL09lEOQO+HKRJcqn3X+vviYBQIQFXDt4-WF5g2xlKfTLe6FFHjAD+n+x7YKLQNpbrNC74ai8D9pb5n7U4BR05ZBK77hIwDBZCowoHIFgB14SqN43Qt4iDt5ZAADEAAYqQXXvQNZK0FYGZHXJTn-igffigYgUQQAJxsHkFyBQgZiEEvC8Ei7i4-CCDQAAEOxAGlp6b8Ak5rDk775KgO7K4oRPyAH3LiEbaSFbJSFk4U5R6-4oDK5KG8BI4QYuTwwoA4BIBoDuBaAgAAi8DnQqDY40CQC3Brz2AgCVhKAWA2DkCSBIBjgzDiCnQahcJgCxDOFWCuGq44AYDRHQZhEayuFoYgAmbnA0BtAQDiAuAyDI4mE0A4Aj5YAWFWFuBAA)

Ohh, this is great. And, I since I was unable to make the modal-container work. I just simply rendered a component using `m()` and It is working.
https://github.com/zelfroster/RSNewWebUI/tree/mail-section

**Comment by zelfroster** [comment]
So, for now @rottencandy, can this PR be approved?

**Review by rottencandy — APPROVED** [review]
Yeah I think we can go ahead with this

**Pull request merged**